frontend/javascripts/admin/dataset/dataset_add_zarr_view.tsx

```diff
@@ -327,13 +343,28 @@ function AddZarrLayer({
     syncDataSourceFields(form, dataSourceEditMode === "simple" ? "advanced" : "simple");
     const datasourceConfigStr = form.getFieldValue("dataSourceJson");
 
-    const { dataSource: newDataSource, report } =
-      !usernameOrAccessKey || !passwordOrSecretKey
-        ? await exploreRemoteDataset([datasourceUrl])
-        : await exploreRemoteDataset([datasourceUrl], {
+    const { dataSource: newDataSource, report } = await (async () => {
+      if (showCredentialsFields) {
+        if (selectedProtocol === "gs") {
+          const credentials =
+            fileList.length > 0 ? await parseCredentials(fileList[0]?.originFileObj) : null;
+          if (credentials) {
+            return exploreRemoteDataset([datasourceUrl], {
+              username: "",
+              pass: JSON.stringify(credentials),
```

Comment:
I tried to find out which type `credentials` has, but it's not strictly specified in wk. The only thing I found was `Record<string, any>` in the `NeuroglancerDatasetConfig`. Is there a more specific definition of the credential type? And is stringifying and passing all of it as the password here, correct?

Reply:
Ah I found the link to the documentation which shows what the credentials file looks like (https://cloud.google.com/iam/docs/creating-managing-service-account-keys?hl=de#creating). In that case, there doesn't need to be a more specific type and it also makes sense to pass all of it as `pass` here, so nevermind :)